db/migrate/20221202130039_fill_project_journals_with_existing_data.rb

```diff
@@ -0,0 +1,99 @@
+class FillProjectJournalsWithExistingData < ActiveRecord::Migration[7.0]
+  def up
+    create_journal_entries_for_projects
+  end
+
+  def down
+    delete_journal_entries_for_projects
+  end
+
+  private
+
+  def create_journal_entries_for_projects
+    sql = <<~SQL.squish
+      WITH project_journals_insertion AS (
+        INSERT INTO project_journals(
+            name,
+            description,
+            public,
+            parent_id,
+            identifier,
+            active,
+            templated
+          )
+        SELECT name,
+          description,
+          public,
+          parent_id,
+          identifier,
+          active,
+          templated
+        FROM projects
+        RETURNING id,
+          identifier
+      ),
+      journals_insertion AS (
+        INSERT into journals (
+            journable_id,
+            journable_type,
+            user_id,
+            created_at,
+            updated_at,
+            version,
+            activity_type,
+            data_type,
+            data_id
+          )
+        SELECT projects.id,
+          'Project',
+          :user_id,
+          projects.created_at,
+          projects.updated_at,
+          1,
+          'projects',
```

Comment:
This will probably break now that #11756 is merged and removes the `activity_type`

Reply:
Indeed.
I took care of giving the migration in #11756 a [version number](https://github.com/opf/openproject/pull/11756/files#diff-b4737f29b8325a24816043c06d5e0dfe65453026223e2d17879988be6d312395) which is before the migrations of this PR, so that I can remove all things related to `activity_type`.

Thanks for merging the other PR and raising the concern.

I'll now squash commits together and everything should be good.

Reply:
Great. Merge at will once you are satisfied.